src/server/pps/server/api_server_integration_test.go

```diff
@@ -137,3 +141,95 @@ func TestRerunPipeline(t *testing.T) {
 	require.Equal(t, int64(0), jobInfo.DataRecovered)
 	require.Equal(t, int64(0), jobInfo.DataFailed)
 }
+
+// TestSidecarMetrics tests that sidecar metrics are available.
+func TestSidecarMetrics(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	t.Parallel()
+	c, namespace := minikubetestenv.AcquireCluster(t)
+
+	inputRepo := "input"
+	require.NoError(t, c.CreateRepo(pfs.DefaultProjectName, inputRepo))
+
+	var pipelineTemplate1 = `{
+		"pipeline": {
+			"project": {
+				"name": "{{.ProjectName | js}}"
+			},
+			"name": "{{.PipelineName | js}}"
+		},
+		"transform": {
+			"cmd": ["cp", "-r", "/pfs/in/foo", "/pfs/out"]
+		},
+		"input": {
+			"pfs": {
+				"project": "default",
+				"repo": "{{.RepoName | js}}",
+				"glob": "/*",
+				"name": "in"
+			}
+		},
+		"autoscaling": false
+	}`
+	tmpl1, err := template.New("pipeline").Parse(pipelineTemplate1)
+	require.NoError(t, err, "template must parse")
+	var buf bytes.Buffer
+	require.NoError(t, tmpl1.Execute(&buf, struct {
+		ProjectName, PipelineName, RepoName string
+	}{pfs.DefaultProjectName, "pipeline1", inputRepo}), "template must execute")
+	_, err = c.PpsAPIClient.CreatePipelineV2(ctx, &pps.CreatePipelineV2Request{
+		CreatePipelineRequestJson: buf.String(),
+	})
+	require.NoError(t, err, "creating pipeline1 must succeed")
+
+	// pipeline2 pulls metrics from pipeline1
+	var pipelineTemplate2 = `{
+		"pipeline": {
+			"project": {
+				"name": "{{.ProjectName | js}}"
+			},
+			"name": "{{.PipelineName | js}}"
+		},
+		"transform": {
+			"image": "curlimages/curl:8.5.0",
+			"cmd": ["sh", "-c", "curl http://default-pipeline1-v1.{{.Namespace}}.svc.cluster.local:9091/metrics > /pfs/out/output"]
```

Comment:
`default-pipeline1-v1` alone should work, I think?  This URL doesn't avoid ndots problems because it ends with `.local` and not `.local.`... (so if you can't resolve the address, it tries default-pipeline1-v1.namespace.svc.cluster.local.namespace, default-pipeline1-v1.namespace.svc, ...)